1-Authentication/sign-in-b2c/README.md

```diff
@@ -0,0 +1,296 @@
+
+### Project Initialization
+
+Create a new Java Maven project and copy the `pom.xml` file from this project, and the `src` folder of this repository.
+
+If you'd like to create a project like this from scratch, you may use [Spring Initializer](https://start.spring.io):
+
+- For **Packaging**, select `Jar`
+- For **Java** select version `11`
+- For **Dependencies**, add the following:
+  - Azure Active Directory
+  - Spring Oauth2 Client
+  - Spring Web
+- Be sure that it comes with Azure SDK version 3.3 or higher. If not, please consider replacing the pre-configured `pom.xml` with the `pom.xml` from this repository.
+
+### ID Token Claims
+
+To extract token details, make use of Spring Security's `AuthenticationPrincipal` and `OidcUser` object in a request mapping. See the [Sample Controller](./src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapp/SampleController.java) for an example of this app making use of ID Token claims.
+
+```java
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+//...
+@GetMapping(path = "/some_path")
+public String tokenDetails(@AuthenticationPrincipal OidcUser principal) {
+    Map<String, Object> claims = principal.getIdToken().getClaims();
+}
+```
+
+### Sign-in and sign-out links
+
+To sign in, you must make a request to the Azure Active Directory sign-in endpoint that is automatically configured by **Azure AD B2C Spring Boot Starter client library for Java**.
+
+```html
+<a class="btn btn-success" href="/oauth2/authorization/{your-sign-up-sign-in-user-flow}">Sign In</a>
+```
+
+To sign out, you must make POST request to the **logout** endpoint.
+
+```HTML
+<form action="#" th:action="@{/logout}" method="post">
+  <input class="btn btn-warning" type="submit" value="Sign Out" />
+</form>
+```
+
+### Authentication-dependent UI elements
+
+This app has some simple logic in the UI template pages for determining content to display based on whether the user is authenticated or not. For example, the following Spring Security Thymeleaf tags may be used:
+
+```html
+<div sec:authorize="isAuthenticated()">
+  this content only shows to authenticated users
+</div>
+<div sec:authorize="isAnonymous()">
+  this content only shows to not-authenticated users
+</div>
+```
+
+### Protecting routes with AADWebSecurityConfigurerAdapter
+
+By default, this app protects the **ID Token Details** page so that only logged-in users can access it. This app uses configures these routes from the `app.protect.authenticated` property from the `application.yml` file. To configure your app's specific requirements, extend `AADWebSecurityConfigurationAdapter` in one of your classes. For an example, see this app's [SecurityConfig](./src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapp/SecurityConfig.java) class.
```

Comment:
```suggestion
By default, this app protects the **ID Token Details** page so that only logged-in users can access it. This app uses configures these routes from the `app.protect.authenticated` property from the `application.yml` file. To configure your app's specific requirements, extend `WebSecurityConfigurerAdapter` in one of your classes. For an example, see this app's [SecurityConfig](./src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapp/SecurityConfig.java) class.
```